lgc/patch/PatchInOutImportExport.cpp

```diff
@@ -1463,38 +1473,40 @@ void PatchInOutImportExport::visitReturnInst(ReturnInst &retInst) {
 
       // NOTE: We have to export gl_Layer via generic outputs as well.
       if (useLayer) {
-        bool hasLayerExport = true;
         if (nextStage == ShaderStageFragment) {
-          const auto &nextBuiltInUsage = m_pipelineState->getShaderResourceUsage(ShaderStageFragment)->builtInUsage.fs;
-
-          hasLayerExport = nextBuiltInUsage.layer || nextBuiltInUsage.viewIndex;
-        } else if (nextStage == ShaderStageInvalid) {
-          hasLayerExport = false;
-        }
-
-        if (hasLayerExport) {
           unsigned loc = InvalidValue;
-          if (m_shaderStage == ShaderStageCopyShader) {
-            assert(inOutUsage.gs.builtInOutLocs.find(BuiltInLayer) != inOutUsage.gs.builtInOutLocs.end() ||
-                   inOutUsage.gs.builtInOutLocs.find(BuiltInViewIndex) != inOutUsage.gs.builtInOutLocs.end());
-
-            loc = enableMultiView ? inOutUsage.gs.builtInOutLocs[BuiltInViewIndex]
-                                  : inOutUsage.gs.builtInOutLocs[BuiltInLayer];
-          } else {
-            assert(inOutUsage.builtInOutputLocMap.find(BuiltInLayer) != inOutUsage.builtInOutputLocMap.end() ||
-                   inOutUsage.builtInOutputLocMap.find(BuiltInViewIndex) != inOutUsage.builtInOutputLocMap.end());
+          Value *layer = nullptr;
+          const auto &nextBuiltInUsage = m_pipelineState->getShaderResourceUsage(ShaderStageFragment)->builtInUsage.fs;
 
-            loc = enableMultiView ? inOutUsage.builtInOutputLocMap[BuiltInViewIndex]
-                                  : inOutUsage.builtInOutputLocMap[BuiltInLayer];
+          if (nextBuiltInUsage.layer) {
+            if (m_shaderStage == ShaderStageCopyShader) {
+              assert(inOutUsage.gs.builtInOutLocs.find(BuiltInLayer) != inOutUsage.gs.builtInOutLocs.end());
```

Comment:
You can do `assert(is_contained(set, element))`. Also everywhere else.

Reply:
Currently this format is heavily used, I think another change might be better. Thanks.